clash_lib/src/proxy/tuic/handle_task.rs

```diff
@@ -0,0 +1,155 @@
+use bytes::Bytes;
+use quinn::ZeroRttAccepted;
+
+use anyhow::Result;
+use std::time::Duration;
+use tokio::time;
+use tuic::Address;
+use tuic_quinn::{Connect, Packet};
+
+use crate::session::SocksAddr as ClashSocksAddr;
+use crate::proxy::datagram::UdpPacket;
+
+use super::types::{TuicConnection, UdpRelayMode};
+
+impl TuicConnection {
+    pub async fn tuic_auth(self, zero_rtt_accepted: Option<ZeroRttAccepted>) {
```

Comment:
do you really need to consume `self` here? https://docs.rs/tuic-quinn/0.1.0/tuic_quinn/struct.Connection.html#method.authenticate

Reply:
Consuming self here mainly aims make `self` live longer, because tokio::spawn requires 'static